raiden-cli/src/index.ts

```diff
@@ -231,98 +44,71 @@ async function main() {
-      },
-      mint: {
-        alias: 'm',
-        type: 'boolean',
-        desc: 'Mint amount to deposit',
-        implies: 'deposit',
-      },
-      transfer: {
-        type: 'string',
-        desc: 'Transfer this amount to target',
-        implies: 'token',
-      },
-      target: {
-        type: 'string',
-        desc: 'Transfer to this address',
-        implies: 'transfer',
-      },
-      serve: {
+      port: {
         type: 'number',
+        default: 5001,
         desc: 'Serve HTTP API on given port',
       },
     })
     .env('RAIDEN')
     .help().argv;
+}
 
-  // eslint-disable-next-line prefer-const
-  let stopSub: ReturnType<typeof raiden.events$.subscribe>;
-  const raidenStop = () => {
-    if (stopSub) stopSub.unsubscribe();
-    if (server?.listening) server.close();
-    if (!raiden?.started) process.exit(1);
-    log.info('Stopping...');
-    raiden.stop();
-  };
-  process.on('SIGINT', raidenStop);
-  process.on('SIGTERM', raidenStop);
-
-  if (!argv.password) {
-    // argv.password = inquirer.prompt('Private Key Password: ', { hideEchoBack: true });
-    argv.password = (
-      await inquirer.prompt<{ password: string }>([
-        { type: 'password', name: 'password', message: 'Private Key Password:', mask: '*' },
-      ])
-    ).password;
-  }
+async function askUserForPassword(): Promise<string> {
+  const userInput = await inquirer.prompt<{ password: string }>([
+    { type: 'password', name: 'password', message: 'Private Key Password:', mask: '*' },
+  ]);
+  return userInput.password;
+}
 
-  const wallet = await Wallet.fromEncryptedJson(
-    await fs.readFile(argv.privateKey, 'utf-8'),
-    argv.password,
-  );
-  log = logging.getLogger(`cli:${wallet.address}`);
-  log.info('Address:', wallet.address);
+async function getWallet(privateKeyPath: string, password: string): Promise<Wallet> {
+  const encryptedKey = await fs.readFile(privateKeyPath, 'utf-8');
+  return await Wallet.fromEncryptedJson(encryptedKey, password);
+}
 
-  const localStorage = new LocalStorage(argv.store);
+function createLocalStorage(name: string): LocalStorage {
+  const localStorage = new LocalStorage(name);
   Object.assign(globalThis, { localStorage });
+  return localStorage;
+}
 
-  const defaultConfig = {
-    matrixServer: 'https://raidentransport.test001.env.raiden.network',
-    pfs: 'https://pfs.raidentransport.test001.env.raiden.network',
-    pfsSafetyMargin: 1.1,
-    caps: { noDelivery: true, webRTC: true },
-  };
-  raiden = await Raiden.create(argv.ethNode, wallet.privateKey, localStorage, undefined, {
-    ...defaultConfig,
-    ...argv.config,
-  });
+function initLogging(address: string): void {
+  log = logging.getLogger(`cli:${address}`);
+  log.info('Address:', address);
+}
 
-  const init = raidenInit(argv);
-  raiden.start();
-  // if events$ completes, it means raiden shut down
-  stopSub = raiden.events$.subscribe(undefined, raidenStop, raidenStop);
+function shutdown(): void {
```

Comment:
Careful with shutdown, you're losing something here: currently,  we use `raiden?.started` to check if we already stopped it once (i.e. it's a second `Ctrl+C`), and force exit on it. It means on first SIGINT we call stop and try to wait for it to exit gracefully (node to finish any pending http request, scheduled timeout handler, etc), and if it's slow/stuck, the user can hit it again and it exit(1). Your shutdown here may lose that.

Reply:
Oh, thanks that you caught that! I'll need to fix that. :+1:

Reply:
I pushed a fix for that. Could you please verify that it is correct? Just to be sure. Thanks!

Reply:
Looks right now, besides the RaidenService discussion. Thanks!